File: Leetcode/1672_richest_customer_wealth/solution.py
"""
  link: https://leetcode.com/problems/richest-customer-wealth/submissions/
	memory: O(1)
	time: O(n * m), row * column
	tag: Array, Matrix
"""

accounts = [[2, 8, 7], [7, 1, 3], [1, 9, 5]]


def maximumWealth(accounts):
    s = 0
    for i in range(0, len(accounts)):
        t = 0
        for j in range(0, len(accounts[i])):
            t += accounts[i][j]
        s = max(s, t)

    return s


print(maximumWealth(accounts))

File: Leetcode/1672_richest_customer_wealth/solution.go
/*
	link: https://leetcode.com/problems/richest-customer-wealth/submissions/
	memory: O(1)
	time: O(n * m), row * column
	tag: Array, Matrix
*/
package main

import (
	"fmt"
)

func maximumWealth(accounts [][]int) int {
	 s := 0;

	for i := 0; i < len(accounts); i++ {
		t := 0
		for j := 0; j < len(accounts[i]); j++ {
			t += accounts[i][j]
		}

		if(t > s) {
			s = t;
		}
		
	}

	return s;
}

func main() {
	s := [][]int {
		{1, 2, 3},
		{10, 2, 1},
	}


	fmt.Println(maximumWealth(s)) 
}